lib/active_merchant/billing/gateways/commerce_hub.rb

```diff
@@ -171,10 +172,7 @@ def add_billing_address(post, payment, options)
         return unless billing = options[:billing_address]
 
         billing_address = {}
-        if payment.is_a?(CreditCard)
-          billing_address[:firstName] = payment.first_name if payment.first_name
-          billing_address[:lastName] = payment.last_name if payment.last_name
-        end
+        name_from_payment(billing_address, payment) || name_from_address(billing_address, options) if payment.is_a?(CreditCard)
```

Comment:
I think the order of this `or` statement needs to be swapped - if the  billing address is provided, the idea is that those values override what might be stored on the card

Reply:
I also wonder if the `if payment.is_a?(CreditCard)` is necessary here or if it should be a guard clause on `name_from_payment` instead?

Or better yet, should `name_from_payment` just check if `payment.respond_to?(:first_name)` (and last name)? This gateway also supports ApplePay and GooglePay, so that solution might be more flexible and ensure that we aren't skipping logic that could apply to non-credit card payments

Reply:
ahh i see! I didnt catch that we were prioritizing the billing_address first! Also good point on the googlepay/apple pay thought!